datumaro/plugins/open_images_format.py

```diff
@@ -0,0 +1,155 @@
+    def _open_csv_annotation(self, file_name):
+        absolute_path = osp.join(self._dataset_dir, _OpenImagesPath.ANNOTATIONS_DIR, file_name)
+
+        with open(absolute_path, 'r', newline='') as f:
+            yield csv.DictReader(f)
+
+    def _glob_annotations(self, pattern):
+        return glob.glob(osp.join(
+            glob.escape(self._dataset_dir), _OpenImagesPath.ANNOTATIONS_DIR, pattern))
+
+    def _load_categories(self):
+        label_categories = LabelCategories()
+
+        with self._open_csv_annotation('oidv6-class-descriptions.csv') as class_description_reader:
+            for class_description in class_description_reader:
+                label_categories.add(class_description['LabelName'])
+
+        self._categories[AnnotationType.label] = label_categories
+
+        self._load_label_category_parents()
+
+    def _load_label_category_parents(self):
+        label_categories = self._categories[AnnotationType.label]
+
+        hierarchy_path = osp.join(
+            self._dataset_dir, _OpenImagesPath.ANNOTATIONS_DIR, 'bbox_labels_600_hierarchy.json')
+
+        with open(hierarchy_path, 'rb') as hierarchy_file:
+            root_node = json.load(hierarchy_file)
+
+        def set_parents_from_node(node, category):
+            for child_node in node.get('Subcategory', []):
+                _, child_category = label_categories.find(child_node['LabelName'])
+
+                if category is not None and child_category is not None:
+                    child_category.parent = category.name
+
+                set_parents_from_node(child_node, child_category)
+
+        _, root_category = label_categories.find(root_node['LabelName'])
+        set_parents_from_node(root_node, root_category)
+
+    def _load_items(self):
+        items_by_id = {}
+
+        def load_from(path):
+            with self._open_csv_annotation(path) as image_reader:
+                for image_description in image_reader:
+                    image_id = image_description['ImageID']
+                    assert _RE_IMAGE_ID.fullmatch(image_id)
+                    assert image_id not in items_by_id
+
+                    subset = image_description['Subset']
+                    assert subset in {'test', 'train', 'validation'}
+
+                    items_by_id[image_id] = DatasetItem(
+                        id=image_id,
+                        image=osp.join(self._dataset_dir, f'images/{subset}/{image_id}.jpg'),
+                        subset=subset,
+                    )
+
+        # It's preferable to load the combined image description file,
+        # because it contains descriptions for training images without human-annotated labels
+        # (the file specific to the training set doesn't).
+        # However, if it's missing, we'll try loading subset-specific files instead, so that
+        # this exporter can be used on individual subsets of the dataset.
+        try:
+            load_from(_OpenImagesPath.FULL_IMAGE_DESCRIPTION_NAME)
```

Comment:
> I feel like I'd want to pass an absolute path here. It would be much more self-explanatory.

That would involve duplicating the code that joins the annotation file name with the path of the `annotations` directory. I instead tried to make it clearer by renaming the `load_from` parameter to `annotation_name` (as well as ensuring that it is always a name).

> Maybe, I'd also had this function with a return value.

I don't think that would be any better. The function itself would be longer (as it would need to collect the results into a temporary dictionary), the usages would be longer and more repetitive (since we would need to merge the return values into `items_by_id`), and the runtime would take longer (since the items would first be placed into the temporary dictionary and then merged into another dictionary).